data-raw/workflow/workflow_union_3_years_in_one_table_pseudonymised.sql

```diff
@@ -0,0 +1,232 @@
+-- To run from DALL_REF
+
+drop table int646_base_20200401_20230331_pseudo purge;
+create table int646_base_20200401_20230331_pseudo nologging compress for query low as
+
+select 
+    rawtohex(standard_hash(to_char(nhs_no), 'SHA256')) nhs_no_hash,
+    base.*,
+    opdd.lsoa_code
+from dall_ref.int646_base_20200401_20230331 base
+left join (
+    select
+      postcode,
+      census_lower as lsoa_code
+    from (
+      select
+        q01.*,
+        rank() over (partition by postcode order by year_month desc) as rank
+      from (
+        select
+          year_month,
+          date_of_termination,
+          county_code,
+          local_auth_district_code,
+          ward_code,
+          country_code,
+          sha_code,
+          region,
+          local_office,
+          ccg_code,
+          pcon_code,
+          oseast1m,
+          osnrth1m,
+          census_lower,
+          census_middle,
+          postcode
```

Comment:
I initially started to remove any columns that were not needed, but realised it firstly does not matter to end result and secondly there is a risk of removing something that is actually needed in an intermediate step.